cmd/oras/root/manifest/index/update.go

```diff
@@ -0,0 +1,235 @@
+	}
+	printUpdateStatus(status.IndexPromptFetched, opts.Reference, string(desc.Digest), opts.Printer)
+	var index ocispec.Index
+	if err := json.Unmarshal(content, &index); err != nil {
+		return ocispec.Index{}, err
+	}
+	return index, nil
+}
+
+func addManifests(ctx context.Context, manifests []ocispec.Descriptor, target oras.ReadOnlyTarget, opts updateOptions) ([]ocispec.Descriptor, error) {
+	for _, manifestRef := range opts.addArguments {
+		printUpdateStatus(status.IndexPromptFetching, manifestRef, "", opts.Printer)
+		desc, content, err := oras.FetchBytes(ctx, target, manifestRef, oras.DefaultFetchBytesOptions)
+		if err != nil {
+			return nil, fmt.Errorf("could not find the manifest %s: %w", manifestRef, err)
+		}
+		if !descriptor.IsManifest(desc) {
+			return nil, fmt.Errorf("%s is not a manifest", manifestRef)
+		}
+		printUpdateStatus(status.IndexPromptFetched, manifestRef, string(desc.Digest), opts.Printer)
+		if descriptor.IsImageManifest(desc) {
+			desc.Platform, err = getPlatform(ctx, target, content)
+			if err != nil {
+				return nil, err
+			}
+		}
+		manifests = append(manifests, desc)
+		printUpdateStatus(status.IndexPromptAdded, manifestRef, string(desc.Digest), opts.Printer)
+	}
+	return manifests, nil
+}
+
+func mergeIndexes(ctx context.Context, manifests []ocispec.Descriptor, target oras.ReadOnlyTarget, opts updateOptions) ([]ocispec.Descriptor, error) {
+	for _, indexRef := range opts.mergeArguments {
+		printUpdateStatus(status.IndexPromptFetching, indexRef, "", opts.Printer)
+		desc, content, err := oras.FetchBytes(ctx, target, indexRef, oras.DefaultFetchBytesOptions)
+		if err != nil {
+			return nil, fmt.Errorf("could not find the index %s: %w", indexRef, err)
+		}
+		if !descriptor.IsIndex(desc) {
+			return nil, fmt.Errorf("%s is not an index", indexRef)
+		}
+		printUpdateStatus(status.IndexPromptFetched, indexRef, string(desc.Digest), opts.Printer)
+		var index ocispec.Index
+		if err := json.Unmarshal(content, &index); err != nil {
+			return nil, err
+		}
+		manifests = append(manifests, index.Manifests...)
+		printUpdateStatus(status.IndexPromptMerged, indexRef, string(desc.Digest), opts.Printer)
+	}
+	return manifests, nil
+}
+
+func removeManifests(ctx context.Context, manifests []ocispec.Descriptor, target oras.ReadOnlyTarget, opts updateOptions) ([]ocispec.Descriptor, error) {
+	// create a set of digests to speed up the remove
+	digestToRemove := make(map[digest.Digest]bool)
+	for _, manifestRef := range opts.removeArguments {
+		if !contentutil.IsDigest(manifestRef) {
+			return nil, fmt.Errorf("%s is not a digest", manifestRef)
+		}
+		digestToRemove[digest.Digest(manifestRef)] = false
+	}
+	return doRemoveManifests(manifests, digestToRemove, opts.Printer, opts.Reference)
+}
+
+func doRemoveManifests(originalManifests []ocispec.Descriptor, digestSet map[digest.Digest]bool, printer *output.Printer, indexRef string) ([]ocispec.Descriptor, error) {
+	manifests := []ocispec.Descriptor{}
```

Comment:
```suggestion
	var manifests []ocispec.Descriptor
```

Reply:
Doing this will initialize `manifests` to be `nil` instead of a non-nil empty slice. This will cause error when all items of the original slice is deleted and nothing is copied to the new slice. The `manifests` field of the updated `index` will be `manifests: nil` instead of `manifests: []`.